Content.Server/Explosion/EntitySystems/ExplosionSystem.GridMap.cs

```diff
@@ -0,0 +1,374 @@
+                {
+                    // this is purely a diagonal edge tile
+                    var newIndex = new Vector2i((int) MathF.Floor(center.X), (int) MathF.Floor(center.Y));
+                    if (!transformedEdges.TryGetValue(newIndex, out var data))
+                    {
+                        data = new();
+                        transformedEdges[newIndex] = data;
+                    }
+
+                    data.BlockingGridEdges.Add(new(default, null, center, angle, tileSize));
+                    continue;
+                }
+
+                // Instead of just mapping the center of the tile, we map for points on that tile. This is basically a
+                // shitty approximation to doing a proper check to get all space-tiles that intersect this grid tile.
+                // Not perfect, but works well enough.
+                transformedTiles.Clear();
+                transformedTiles.Add(new((int) MathF.Floor(center.X + x), (int) MathF.Floor(center.Y + y)));  // center of tile, offset by (0.25, 0.25) in tile coordinates
+                transformedTiles.Add(new((int) MathF.Floor(center.X - y), (int) MathF.Floor(center.Y + x)));  // center offset by (-0.25, 0.25)
+                transformedTiles.Add(new((int) MathF.Floor(center.X - x), (int) MathF.Floor(center.Y - y)));  // offset by (-0.25, -0.25)
+                transformedTiles.Add(new((int) MathF.Floor(center.X + y), (int) MathF.Floor(center.Y - x)));  // offset by (0.25, -0.25)
+
+                foreach (var newIndices in transformedTiles)
+                {
+                    if (!transformedEdges.TryGetValue(newIndices, out var data))
+                    {
+                        data = new();
+                        transformedEdges[newIndices] = data;
+                    }
+                    data.BlockingGridEdges.Add(new(tile, gridToTransform, center, angle, tileSize));
+                }
+            }
+        }
+
+        if (referenceGrid == null)
+            return (transformedEdges, tileSize);
+
+        // finally, we also include the blocking tiles from the reference grid.
+
+        if (_gridEdges.TryGetValue(referenceGrid.Value, out var localEdges))
+        {
+            foreach (var (tile, dir) in localEdges)
+            {
+                // grids cannot overlap, so tile should NEVER be an existing entry.
+                var data = new BlockedSpaceTile();
+                transformedEdges[tile] = data;
+                
+                data.UnblockedDirections = AtmosDirection.Invalid; // all directions are blocked automatically.
+
+                if ((dir & NeighborFlag.Cardinal) == 0)
+                    data.BlockingGridEdges.Add(new(default, null, ((Vector2) tile + 0.5f) * tileSize, 0, tileSize));
+                else
+                    data.BlockingGridEdges.Add(new(tile, referenceGrid.Value, ((Vector2) tile + 0.5f) * tileSize, 0, tileSize));
+            }
+        }
```

Comment:
Yeah its all pretty sussy. But IMO not as sussy as just not having multi grid explosions.

Probably the best / most thorough way of reviewing this might be for other people to just try solve multi-grid explosions independently and seeing what they come up with, cause it might be much better.